packages/Search/examples/bvh_driver/bvh_driver.cpp

```diff
@@ -259,26 +260,44 @@ int main( int argc, char *argv[] )
 {
     KokkosScopeGuard guard( argc, argv );
 
-    bool const throw_exceptions = false;
-    bool const recognise_all_options = false;
-    Teuchos::CommandLineProcessor clp( throw_exceptions,
-                                       recognise_all_options );
+    namespace bpo = boost::program_options;
+    bpo::options_description desc( "Allowed options" );
     int n_values = 50000;
     int n_queries = 20000;
     int n_neighbors = 10;
     int buffer_size = 0;
     std::string source_pt_cloud = "filled_box";
     std::string target_pt_cloud = "filled_box";
-    clp.setOption( "values", &n_values, "number of indexable values (source)" );
-    clp.setOption( "queries", &n_queries, "number of queries (target)" );
-    clp.setOption( "neighbors", &n_neighbors,
-                   "desired number of results per query" );
-    clp.setOption( "buffer", &buffer_size,
-                   "size for buffer optimization in radius search" );
-    clp.setOption( "source-point-cloud-type", &source_pt_cloud,
-                   "shape of the source point cloud" );
-    clp.setOption( "target-point-cloud-type", &target_pt_cloud,
-                   "shape of the target point cloud" );
+    // clang-format off
+    desc.add_options()
+        ( "help", "produce help message" )
+        ( "values", bpo::value<int>(&n_values)->default_value(50000), "number of indexable values (source)" )
+        ( "queries", bpo::value<int>(&n_queries)->default_value(20000), "number of queries (target)" )
+        ( "neighbors", bpo::value<int>(&n_neighbors)->default_value(10), "desired number of results per query" )
+        ( "buffer", bpo::value<int>(&buffer_size)->default_value(0), "size for buffer optimization in radius search" )
+        ( "source-point-cloud-type", bpo::value<std::string>(&source_pt_cloud)->default_value("filled_box"), "shape of the source point cloud"  )
+        ( "target-point-cloud-type", bpo::value<std::string>(&target_pt_cloud)->default_value("filled_box"), "shape of the target point cloud"  )
+    ;
+    // clang-format on
+    bpo::variables_map vm;
+    bpo::store( bpo::parse_command_line( argc, argv, desc ), vm );
+    bpo::notify( vm );
+
+    if ( vm.count( "help" ) )
+    {
+        std::cout << desc << "\n";
+        int ac = 2;
+        char *av[] = {(char *)"ignored", (char *)"--help"};
+        // benchmark::Initialize() calls exit(0) when `--help` so register
+        // Kokkos::finalize() to be called on normal program termination.
```

Comment:
Move the comment 2 lines above, and update it with something like "The full list of help options consist of the boost program options above, and options provided by benchmark. We need to call benchmark::Initialize to produce the latter."

Reply:
You mean add?  What you suggest is trivial.  The comment I have is a "gotcha"

Reply:
It was not trivial to me, and took a while to understand why one needs to call `benchmark::Initialize` in the first place.

Reply:
Done.